arroyo/processing/processor.py

```diff
@@ -218,24 +261,36 @@ def _run_once(self) -> None:
                             logger.info(
                                 "Paused for longer than %d seconds", LOG_THRESHOLD_TIME
                             )
+
+                        if paused_duration is not None:
+                            self.__metrics_buffer.add_pause_time(paused_duration)
```

Comment:
This will be overcounting drastically.
```
paused_duration = current_time - self.__paused_timestamp
```
or
```
paused_duration = current_time - self.__last_log_timestamp
```


So, as long as the consumer stays paused, `self.__paused_timestamp` does not increase while `current_time` does, so every time you get to this line you will log the entire time you have been paused since the last time logger.info was called, not the time passed since the last iteration you called `add_pause_time`. 

`self.__last_log_timestamp` is meant to avoid logging a line hundreds of times per second as, when you are paused this is a tight loop. The processor keeps calling submit, and it keeps receiving the exception.

Reply:
Oops, this was supposed to go in the if statement above. Thanks!